core/templates/directives/focus-on.directive.ts

```diff
@@ -16,10 +16,57 @@
  * @fileoverview FocusOn Directive (not associated with reusable
  * components.)
  * NB: Reusable component directives should go in the components/ folder.
+ *
+ * NOTE TO DEVELOPERS: Please make sure that any code changes are done for BOTH
+ * the directives in this file.
+ * There are two attribute directives in this file. One for AngularJS code and
+ * the other for Angular code.
+ * Attribute directives can't be shared at between AngularJS and Angular
+ * (similar to how pipes/filters can't be). The best that can be done in this
+ * situation is shifting common code in a function and reusing those functions
+ * or just keeping the repeated code in the file.
+ * The old code can be safely removed when the directive is no longer used in
+ * AngularJS codebase.
  */
 
+import { Directive, ElementRef, Input, OnDestroy } from '@angular/core';
+
 import { Subscription } from 'rxjs';
 
+import { AppConstants } from 'app.constants';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+
+@Directive({
+  selector: '[oppiaFocusOn]'
```

Comment:
Are we using this new directive anywhere? Also, do we need to write a test for this new directive?

Reply:
@DubeySandeep, these changes have been tested by Eric (in #11034), the focus-on directive is working as expected. Should we merge this PR or just let these changes go in through Eric's PR. He has pulled my changes so it shouldn't raise a conflict.